pkg/util/inplaceupdate/inplace_update_defaults.go

```diff
@@ -149,18 +141,27 @@ func defaultCalculateInPlaceUpdateSpec(oldRevision, newRevision *apps.Controller
 // If the imageID in containerStatuses has not been changed, we assume that kubelet has not updated
 // containers in Pod.
 func DefaultCheckInPlaceUpdateCompleted(pod *v1.Pod) error {
+	if _, isInGraceState := appspub.GetInPlaceUpdateGrace(pod); isInGraceState {
+		return fmt.Errorf("still in grace period of in-place update")
+	}
+
 	inPlaceUpdateState := appspub.InPlaceUpdateState{}
```

Comment:
Is this here for compatibility with the old logic, when runtimeContainerMetaSet!=nil, there should be no need to parse anymore?

Reply:
Have moved this behind runtimeContainerMetaSet. So that if we check container hashes are consistent, it will not unmarshal inPlaceUpdateState.